src/otg/dataset/dataset.py

```diff
@@ -58,7 +60,7 @@ def schema(self: Dataset) -> StructType:
 
     @classmethod
     @abstractmethod
-    def get_schema(cls: type[Dataset]) -> StructType:
+    def get_schema(cls: type[Self]) -> StructType:
```

Comment:
What is the advantage of using `Self` from the typing_extension library over just adding the name of the class?

Reply:
If this practice is better, shouldn't we change everywhere?

Reply:
The particularity here is that `get_schema` is called from the child classes of Dataset.
Here it is saying that cls is not just any Dataset type, but the type of the child class that is currently being defined. 
For example, StudyLocus is technically of type Dataset, but mypy would complain saying that the returned StructType is generic to the Dataset class, not StudyLocus's.